extensions/amp-accordion/0.1/amp-accordion.js

```diff
@@ -50,19 +66,63 @@ class AmpAccordion extends AMP.BaseElement {
       header.setAttribute('role', 'tab');
       content.classList.add('-amp-accordion-content');
       content.setAttribute('role', 'tabpanel');
-      content.setAttribute(
-          'aria-expanded', section.hasAttribute('expanded').toString());
       let contentId = content.getAttribute('id');
       if (!contentId) {
         contentId = this.element.id + '_AMP_content_' + index;
         content.setAttribute('id', contentId);
       }
+      if (this.currentState_[contentId]) {
+        section.setAttribute('expanded', '');
+      } else if (this.currentState_[contentId] == false) {
+        section.removeAttribute('expanded');
+      }
+      content.setAttribute(
+          'aria-expanded', section.hasAttribute('expanded').toString());
       header.setAttribute('aria-controls', contentId);
       header.addEventListener('click', boundOnHeaderClick_);
     });
   }
 
   /**
+   * Generate a sessionStorage Key based on amp-accordion element id.
+   * @return {string}
+   * @private
+   */
+  getSessionStorageKey_() {
+    const id_ = this.element.id;
+    const url = removeFragment(this.win_.location.href);
+    return `amp-${id_}-${url}`;
+  }
+
+  /**
+   * Get previous state from sessionStorage.
+   * @return {!Object|null}
+   * @private
+   */
+  getSessionState_() {
+    try {
+      const sessionStr = this.win_./*OK*/sessionStorage.getItem(this.id_);
```

Comment:
Not the same thing (`sessionStorage` is not durable). But this should probably go into a shared file.